lib/components/bm-confirm-location-on-map.js

```diff
@@ -0,0 +1,148 @@
+/* @flow */
+'use strict'
+
+const geolocation = require('@blinkmobile/geolocation')
+
+/* :: import type {
+  AngularCallbackOptions, Coordinates, NgModelController
+} from '../../types.js' */
+
+const mod = require('../module.js')
+const utils = require('../utils.js')
+
+class BmConfirmLocationOnMapController {
+  /* :: static $inject : string[] */
+
+  // internal use
+
+  /* :: $digest: () => void */
+  /* :: coords: ?Coordinates */
+  /* :: isEditing: boolean */
+
+  // public attributes (after casts / checks)
+
+  /* :: ngDisabled: boolean */
+  /* :: ngModel: NgModelController */
+  /* :: ngReadonly: boolean */
+
+  constructor ($scope /* : Object */) {
+    this.$digest = $scope.$digest.bind($scope)
+    this.coords = null
+    this.isEditing = false
+  }
+
+  $onInit () {}
+
+  $onChanges () {
+    this.ngDisabled = utils.parseBooleanAttribute(this.ngDisabled)
+    this.ngReadonly = utils.parseBooleanAttribute(this.ngReadonly)
+    this.coords = this.ngModel.$viewValue
+  }
+
+  onCancel (event /* : Event */) {
+    this.coords = this.ngModel.$viewValue
+    this.isEditing = false
+  }
+
+  onChange (value /* : ?Coordinates */) {
+    this.coords = value
+  }
+
+  onClear (event /* : Event */) {
+    this.coords = null
+    this.ngModel.$setViewValue(this.coords, event)
+  }
+
+  onConfirm (event /* : Event */) {
+    this.isEditing = false
+    this.ngModel.$setViewValue(this.coords, event)
```

Comment:
I initially used `$rollbackViewValue()` and `$commitViewValue()`, but they didn't actually have any effect. And when I read the documentation, it seemed like this functionality was mostly used for binding behaviours that were asynchronous. Perhaps if I was debouncing or something, this would presumably be the necessary API.